src/libstd/fs.rs

```diff
@@ -663,15 +623,17 @@ impl Permissions {
     }
 }
 
-#[unstable(feature = "file_type", reason = "recently added API")]
 impl FileType {
```

Comment:
It's somewhat of a fine line between being able to write cross-platform code while still understand what is indeed cross platform. If we were to have an exhaustive enum which is the union of all file types on all platforms, it's suddenly not clear what file types come up on which platforms. Which is to say that it's not clear to me that having a union is indeed more cross platform as you're probably still going to have platform-specific code handling various file types on various platforms.